src/algorithms/reco/JetReconstruction.cc

```diff
@@ -23,31 +23,32 @@ namespace eicrecon {
     m_log->trace("Initialized");
   }
 
-  std::vector<edm4eic::ReconstructedParticle*> JetReconstruction::execute(
+  edm4eic::ReconstructedParticleCollection* JetReconstruction::execute(
     const std::vector<const edm4hep::LorentzVectorE*> momenta) {
 
     // Store the jets
-    std::vector<edm4eic::ReconstructedParticle*> jets_edm;
+    edm4eic::ReconstructedParticleCollection* jet_collection = new edm4eic::ReconstructedParticleCollection();
 
     // Skip empty
     if (momenta.empty()) {
       m_log->trace("  Empty particle list.");
-      return jets_edm;
+      return jet_collection;
     }
 
     m_log->trace("  Number of particles: {}", momenta.size());
 
     // Particles for jet reconstrution
     std::vector<PseudoJet> particles;
     for (const auto &mom : momenta) {
+
       // Only cluster particles within the given pt Range
       if ((mom->pt() > m_minCstPt) && (mom->pt() < m_maxCstPt)) {
```

Comment:
May be for another PR to address, but logically, I'm not sure we want to impose the particle / track pT cuts here as we likely want them to be different depending on the input. We usually place some threshold on tracks (and calorimeter clusters) based on the tracker performance, but place a much lower pT threshold on stable MC particles.

Reply:
Definitely! I was thinking on addressing this in another PR, but in making the options for both factories configurable, we'll also be able to set the kinematic thresholds differently between the reconstructed and generated jets.